### ISSUE TITLE
DCGAN/SNGAN architecture with discriminator conditioning

### ISSUE THREAD

**Issue by ariel415el** [Zhendong-Wang/Diffusion-GAN#13]
Hi, what architecture did you use for the DCGAN/SNGAN experiments in the supplementary?

did you just concatenate learned t-embedding to the discriminator input like here ?

```
class Discriminator(nn.Module):
    def __init__(self):
        super(Discriminator, self).__init__()

        self.label_embedding = nn.Embedding(opt.n_classes, opt.n_classes)

        self.model = block(in_dim=int(np.prod(img_shape)) + opt.n_classes)

    def forward(self, img, labels):
        # Concatenate label embedding and image to produce input
        d_in = torch.cat((img.view(img.size(0), -1), self.label_embedding(labels)), -1)
        validity = self.model(d_in)
        return validity
```

**Comment by Zhendong-Wang** [comment]
We merge the DCGAN/SNGAN into the stylegan2 framework for training. t-embedding is concatenated with the image feature after conv layers. In your case, I think it is 
```
out = self.model(img)
out = torch.cat((out.view(out.shape[0], -1), t_embedding), dim=1)
validity = self.adv_layer(out)
```

**Comment by ArielReplicate** [comment]
Oh I see.
Is self.adv_layer a binary linear classifier?

**Comment by ariel415el** [comment]
Oh I see.
Is self.adv_layer a binary linear classifier?

**Comment by Zhendong-Wang** [comment]
Yes, it is a MLP model but not a binary linear classifier. In our case, the discriminator output the logits while not the probability. 

```
gen_logits = self.run_D(gen_img, gen_c, sync=False)
loss_Gmain = torch.nn.functional.softplus(-gen_logits)  # -log(sigmoid(gen_logits))
```

If you are interested in playing around on DCGAN, you can send me an email. I will share with you the code for that. The code is not cleaned and mighe be a little messy, lol.

**Comment by ariel415el** [comment]
Thanks,  I've sent you a an email to your _@columbia.edu_ address

**Comment by Zhendong-Wang** [comment]
Sent.

**Comment by kongj5** [comment]
can you share me a copy of the diffusion training code for DCGAN, I've sent you a an email to your @columbia.edu address

**Comment by kongj5** [comment]
can you share me a copy of the diffusion training code for DCGAN,

**Comment by kongj5** [comment]
Sorry, I don't know your email address, mine is 1815348757@qq.com

**Comment by Zhendong-Wang** [comment]
Sure!

**Comment by Sarah-2021-scu** [comment]
Hey @Zhendong-Wang, could you please share the code of DCGAN, my email is sarah2024in@gmail.com. 
Thanks in advance
